File: storage/data_transpiled/at_coder/agc032/B/4711258.java
public static void print ( String input ) {
  int N = Integer . parseInt ( input ) ;
  int M = ( N % 2 == 0 ? N * ( N - 1 ) / 2 - N / 2 : N * ( N - 1 ) / 2 - ( N - 1 ) / 2 ) ;
  int s = N % 2 == 0 ? N + 1 : N ;
  String ans = Integer . toString ( M ) + '\n' + StringUtils . join ( i , j ) ;
  for ( int i = 1 ;
  i < N ;
  i ++ ) for ( int j = i + 1 ;
  j <= N ;
  j ++ ) if ( i + j != s ) ans += "\n" ;
  System . out . println ( ans ) ;
}

File: storage/data_original/at_coder/agc032/B/4711258.py
# ??
N = int(input())

# N?????????????? (1, N), (2, N-1), ... ?????????
# N?????????????? (1, N-1), (2, N-2), ... ?????????
M = (
    N * (N - 1) // 2 - N // 2 if N % 2 == 0 else
    N * (N - 1) // 2 - (N - 1) // 2
)
s = N + 1 if N % 2 == 0 else N
ans = str(M) + '\n' + '\n'.join(
    '{} {}'.format(i, j)
    for i in range(1, N)
    for j in range(i + 1, N + 1)
    if i + j != s
)

# ??
print(ans)